From fc1a1b0a507adf0eedeb041a70257ae0b3281b58 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gutyina=20Gerg=C5=91?= <gutyina.gergo.2@gmail.com>
Date: Thu, 24 Aug 2023 14:16:42 +0200
Subject: [PATCH] Focus on the client below the cursor after killing a client

---
 dwm.c | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/dwm.c b/dwm.c
index f1d86b2..1932eb2 100644
--- a/dwm.c
+++ b/dwm.c
@@ -169,6 +169,7 @@ static void focus(Client *c);
 static void focusin(XEvent *e);
 static void focusmon(const Arg *arg);
 static void focusstack(const Arg *arg);
+static void focuscursor();
 static Atom getatomprop(Client *c, Atom prop);
 static int getrootptr(int *x, int *y);
 static long getstate(Window w);
@@ -861,6 +862,19 @@ focusstack(const Arg *arg)
 	}
 }
 
+void focuscursor()
+{
+	Client *c = NULL;
+	int x, y;
+	if (getrootptr(&x, &y))
+	{
+		for (c = selmon->clients; c; c = c->next)
+			if (c == NULL || (c->x <= x && x < c->x+c->w && c->y <= y && y < c->y+c->h))
+				break;
+	}
+	focus(c);
+}
+
 Atom
 getatomprop(Client *c, Atom prop)
 {
@@ -1796,9 +1810,9 @@ unmanage(Client *c, int destroyed)
 		XUngrabServer(dpy);
 	}
 	free(c);
-	focus(NULL);
 	updateclientlist();
 	arrange(m);
+	focuscursor();
 }
 
 void
@@ -2058,7 +2072,7 @@ view(const Arg *arg)
 	selmon->seltags ^= 1; /* toggle sel tagset */
 	if (arg->ui & TAGMASK)
 		selmon->tagset[selmon->seltags] = arg->ui & TAGMASK;
-	focus(NULL);
+	focuscursor();
 	arrange(selmon);
 }
 
-- 
2.41.0

