From 22274d1dcc4ed2e945191451ded35251eef7db0a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gutyina=20Gerg=C5=91?= <gutyina.gergo.2@gmail.com>
Date: Thu, 24 Aug 2023 13:43:46 +0200
Subject: [PATCH] Allow disabling and resizing the border

---
 config.def.h |  5 ++++-
 dwm.c        | 31 +++++++++++++++++++++++++++++++
 2 files changed, 35 insertions(+), 1 deletion(-)

diff --git a/config.def.h b/config.def.h
index 9efa774..61079fa 100644
--- a/config.def.h
+++ b/config.def.h
@@ -1,7 +1,7 @@
 /* See LICENSE file for copyright and license details. */
 
 /* appearance */
-static const unsigned int borderpx  = 1;        /* border pixel of windows */
+static unsigned int borderpx        = 1;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
 static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
@@ -65,6 +65,9 @@ static const Key keys[] = {
 	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
 	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
+	{ MODKEY,                       XK_v,      toggleborder,   {0} },
+	{ MODKEY,                       XK_c,      incborder,      {.i = +1 } },
+	{ MODKEY,                       XK_x,      incborder,      {.i = -1 } },
 	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
 	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
 	{ MODKEY,                       XK_i,      incnmaster,     {.i = +1 } },
diff --git a/dwm.c b/dwm.c
index f1d86b2..1ea34e3 100644
--- a/dwm.c
+++ b/dwm.c
@@ -210,6 +210,9 @@ static void tag(const Arg *arg);
 static void tagmon(const Arg *arg);
 static void tile(Monitor *m);
 static void togglebar(const Arg *arg);
+static void updateborder();
+static void toggleborder(const Arg* arg);
+static void incborder(const Arg* arg);
 static void togglefloating(const Arg *arg);
 static void toggletag(const Arg *arg);
 static void toggleview(const Arg *arg);
@@ -266,6 +269,7 @@ static Clr **scheme;
 static Display *dpy;
 static Drw *drw;
 static Monitor *mons, *selmon;
+static int variable_borderpx;
 static Window root, wmcheckwin;
 
 /* configuration, allows nested code to access above variables */
@@ -1589,6 +1593,7 @@ setup(void)
 		scheme[i] = drw_scm_create(drw, colors[i], 3);
 	/* init bars */
 	updatebars();
+	variable_borderpx = borderpx;
 	updatestatus();
 	/* supporting window for NetWMCheck */
 	wmcheckwin = XCreateSimpleWindow(dpy, root, 0, 0, 1, 1, 0, 0, 0);
@@ -1721,6 +1726,32 @@ togglebar(const Arg *arg)
 	arrange(selmon);
 }
 
+void
+updateborder()
+{
+	for(Monitor* m = mons; m; m = m->next)
+	{
+		for(Client* c = m->clients; c; c = c->next)
+			c->bw = borderpx;
+		arrange(m);
+	}
+}
+
+void
+toggleborder(const Arg* arg)
+{
+	borderpx = borderpx == 0 ? variable_borderpx : 0;
+	updateborder();
+}
+
+void
+incborder(const Arg* arg)
+{
+	variable_borderpx = MAX(0, variable_borderpx + arg->i);
+	borderpx = variable_borderpx;
+	updateborder();
+}
+
 void
 togglefloating(const Arg *arg)
 {
-- 
2.41.0

