Author: Gutyina Gergő <gutyina.gergo.2@gmail.com>
Date:   Wed Nov 23 22:17:43 2022 +0200

diff --git a/dwm.c b/dwm.c
index e5efb6a..7a3d9ca 100644
--- a/dwm.c
+++ b/dwm.c
@@ -211,6 +211,9 @@ static void tag(const Arg *arg);
 static void tagmon(const Arg *arg);
 static void tile(Monitor *m);
 static void togglebar(const Arg *arg);
+static void updateborder();
+static void toggleborder(const Arg* arg);
+static void incborder(const Arg* arg);
 static void togglefloating(const Arg *arg);
 static void toggletag(const Arg *arg);
 static void toggleview(const Arg *arg);
@@ -267,6 +270,7 @@ static Clr **scheme;
 static Display *dpy;
 static Drw *drw;
 static Monitor *mons, *selmon;
+static int variable_borderpx;
 static Window root, wmcheckwin;
 
 /* configuration, allows nested code to access above variables */
@@ -1573,6 +1577,7 @@ setup(void)
 		scheme[i] = drw_scm_create(drw, colors[i], 3);
 	/* init bars */
 	updatebars();
+	variable_borderpx = borderpx;
 	updatestatus();
 	/* supporting window for NetWMCheck */
 	wmcheckwin = XCreateSimpleWindow(dpy, root, 0, 0, 1, 1, 0, 0, 0);
@@ -1703,6 +1708,32 @@ togglebar(const Arg *arg)
 	arrange(selmon);
 }
 
+void
+updateborder()
+{
+	for(Monitor* m = mons; m; m = m->next)
+	{
+		for(Client* c = m->clients; c; c = c->next)
+			c->bw = borderpx;
+		arrange(m);
+	}
+}
+
+void
+toggleborder(const Arg* arg)
+{
+	borderpx = borderpx == 0 ? variable_borderpx : 0;
+	updateborder();
+}
+
+void
+incborder(const Arg* arg)
+{
+	variable_borderpx = MAX(0, variable_borderpx + arg->i);
+	borderpx = variable_borderpx;
+	updateborder();
+}
+
 void
 togglefloating(const Arg *arg)
 {
