From 608ed6797d387bf1d0e6980f717ec79f657492f6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gutyina=20Gerg=C5=91?= <gutyina.gergo.2@gmail.com>
Date: Thu, 24 Aug 2023 17:09:59 +0200
Subject: [PATCH] Use config.def.h as main config file

---
 Makefile | 5 +----
 README   | 2 +-
 dwm.1    | 2 +-
 dwm.c    | 4 ++--
 4 files changed, 5 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index 77bcbc0..d9c6761 100644
--- a/Makefile
+++ b/Makefile
@@ -17,10 +17,7 @@ options:
 .c.o:
 	${CC} -c ${CFLAGS} $<
 
-${OBJ}: config.h config.mk
-
-config.h:
-	cp config.def.h $@
+${OBJ}: config.def.h config.mk
 
 dwm: ${OBJ}
 	${CC} -o $@ ${OBJ} ${LDFLAGS}
diff --git a/README b/README
index 95d4fd0..7a516b2 100644
--- a/README
+++ b/README
@@ -44,5 +44,5 @@ like this in your .xinitrc:
 
 Configuration
 -------------
-The configuration of dwm is done by creating a custom config.h
+The configuration of dwm is done by editing config.def.h
 and (re)compiling the source code.
diff --git a/dwm.1 b/dwm.1
index ddc8321..6388fc3 100644
--- a/dwm.1
+++ b/dwm.1
@@ -153,7 +153,7 @@ Toggles focused window between floating and tiled state.
 .B Mod1\-Button3
 Resize focused window while dragging. Tiled windows will be toggled to the floating state.
 .SH CUSTOMIZATION
-dwm is customized by creating a custom config.h and (re)compiling the source
+dwm is customized by creating a custom config.def.h and (re)compiling the source
 code. This keeps it fast, secure and simple.
 .SH SEE ALSO
 .BR dmenu (1),
diff --git a/dwm.c b/dwm.c
index f1d86b2..7507c79 100644
--- a/dwm.c
+++ b/dwm.c
@@ -16,7 +16,7 @@
  * on each monitor. Each client contains a bit array to indicate the tags of a
  * client.
  *
- * Keys and tagging rules are organized as arrays and defined in config.h.
+ * Keys and tagging rules are organized as arrays and defined in config.def.h.
  *
  * To understand everything else, start reading main().
  */
@@ -269,7 +269,7 @@ static Monitor *mons, *selmon;
 static Window root, wmcheckwin;
 
 /* configuration, allows nested code to access above variables */
-#include "config.h"
+#include "config.def.h"
 
 /* compile-time check if all tags fit into an unsigned int bit array. */
 struct NumTags { char limitexceeded[LENGTH(tags) > 31 ? -1 : 1]; };
-- 
2.41.0

