Author: Gutyina Gergő <gutyina.gergo.2@gmail.com>
Date:   Sun Sep 25 10:37:04 2022 +0200

diff --git a/config.def.h b/config.def.h
index e4a7284..d1e7e0d 100644
--- a/config.def.h
+++ b/config.def.h
@@ -6,7 +6,7 @@
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
 static char *font = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
-static int borderpx = 2;
+static int borderpx = 0;
 
 /*
  * What program is execed by st depends of these precedence rules:
diff --git a/st.c b/st.c
index 493600d..1b6306c 100644
--- a/st.c
+++ b/st.c
@@ -2618,7 +2618,15 @@ drawregion(int x1, int y1, int x2, int y2)
 			continue;
 
 		term.dirty[y] = 0;
-		xdrawline(term.line[y], x1, y, x2);
+		xdrawline(term.line[y], x1, y, x2, term.col);
+	}
+	/* use last row as background */
+	if(y == term.row)
+	{
+		Line line = malloc((x2 - x1) * sizeof(Glyph));
+		for(int x = x1; x < x2; x++)
+			line[x] = (Glyph){ ' ', ATTR_NULL, 0, term.line[y - 1][x].bg };
+		xdrawline(line, x1, y, x2, term.col);
 	}
 }
 
diff --git a/win.h b/win.h
index fc63c65..96b8797 100644
--- a/win.h
+++ b/win.h
@@ -26,7 +26,7 @@ enum win_mode {
 void xbell(void);
 void xclipcopy(void);
 void xdrawcursor(int, int, Glyph, int, int, Glyph);
-void xdrawline(Line, int, int, int);
+void xdrawline(Line, int, int, int, int);
 void xfinishdraw(void);
 void xloadcols(void);
 int xsetcolorname(int, const char *);
diff --git a/x.c b/x.c
index fc09737..6d0688e 100644
--- a/x.c
+++ b/x.c
@@ -869,8 +869,8 @@ xhints(void)
 	sizeh->flags = PSize | PResizeInc | PBaseSize | PMinSize;
 	sizeh->height = win.h;
 	sizeh->width = win.w;
-	sizeh->height_inc = win.ch;
-	sizeh->width_inc = win.cw;
+	sizeh->height_inc = 1;
+	sizeh->width_inc = 1;
 	sizeh->base_height = 2 * borderpx;
 	sizeh->base_width = 2 * borderpx;
 	sizeh->min_height = win.ch + 2 * borderpx;
@@ -1646,7 +1646,7 @@ xstartdraw(void)
 }
 
 void
-xdrawline(Line line, int x1, int y1, int x2)
+xdrawline(Line line, int x1, int y1, int x2, int termcols)
 {
 	int i, x, ox, numspecs;
 	Glyph base, new;
@@ -1672,8 +1672,16 @@ xdrawline(Line line, int x1, int y1, int x2)
 		}
 		i++;
 	}
-	if (i > 0)
-		xdrawglyphfontspecs(specs, base, i, ox, y1);
+	if(i > 0)
+	{
+		xdrawglyphfontspecs(specs, base, i, ox, y1);
+		/* use last column as background */
+		if(x == termcols)
+		{
+			Glyph g = { ' ', ATTR_NULL, 0, line[x - 1].bg };
+			xdrawglyphfontspecs(specs, g, i, x, y1);
+		}
+	}
 }
 
 void
